<launch>
  <!-- General arguments -->
  <arg name="param_file"
       default="$(find wavemap_ros)/config/alma.yaml"
       doc="Name of the file from which to read wavemap's server params."/>
  <arg name="debug_mode"
       default="false"
       doc="Run the mapper in a debugger (gdb)."/>
       
  <!-- Run the wavemap server -->
  <group>
    <arg if="$(arg debug_mode)" name="launch_prefix"
         value="gdb -ex run --args"/>
    <arg unless="$(arg debug_mode)" name="launch_prefix" value=""/>

    <node name="wavemap" pkg="wavemap_ros"
          type="wavemap_server_node"
          launch-prefix="$(arg launch_prefix)">
      <rosparam file="$(arg param_file)"/>
    </node>
  </group>

  <!-- Run the waverider server -->
  <node name="waverider_chomp_planner" pkg="waverider_chomp_planner"
        type="waverider_chomp_planner_node"
        output="screen">
  </node>

</launch>
